R/get_vimc_climate.R

```diff
@@ -0,0 +1,150 @@
+    "ERA5_RH",
+    "ERA5_SH",
+    "PERSIANN"
+  ),
+  admin_level = c(0, 1, 2, 3)
+) {
+  # more checks below
+  checkmate::assert_string(
+    country,
+    min.chars = 2
+  )
+
+  # allow date as string because actually using Date class is annoying
+  if (!checkmate::test_vector(date_range, len = 2)) {
+    cli::cli_abort(
+      "`date_range` must be a 2-element vector; currently has a length of
+      length(date_range)"
+    )
+  }
+  is_good_daterange <- checkmate::test_date(
+    date_range,
+    any.missing = FALSE
+  ) ||
+    checkmate::test_character(
+      date_range
+    )
+  if (!is_good_daterange) {
+    cli::cli_abort(
+      "`date_range` must be either a `<Date>` or `string` vector, but is a
+      {.cls {class(date_range)}}"
+    )
+  } else if (is.character(date_range)) {
+    # convert to Date
+    date_range <- lubridate::as_date(date_range)
+  }
+
+  # NOTE: need to check for sensible date values and throw an error
+  # if data are not available in the range
+
+  # allow only one choice at a time to avoid ambiguous return type
+  data_source <- rlang::arg_match(data_source)
+  admin_level <- as.character(admin_level)
+  admin_level <- rlang::arg_match(admin_level, allowed_admin_levels)
+
+  # convert country to ISO 3166-1 3C country code
+  # note `nomatch = NULL` is preferable but is a special case
+  # in pkg countrycode
+  nchar_country <- as.character(nchar(country))
+  country_iso3c <- switch(
+    nchar_country,
+    "2" = countrycode::countrycode(
+      country,
+      "iso2c",
+      "iso3c",
+      warn = FALSE,
+      nomatch = NA
+    ),
+    "3" = countrycode::countrycode(
+      country,
+      "iso3c",
+      "iso3c",
+      warn = FALSE,
+      nomatch = NA
+    ),
+    countrycode::countryname(
+      country,
+      "iso3c",
+      warn = FALSE,
+      nomatch = NA
+    )
+  )
+
+  if (is.na(country_iso3c)) {
+    cli::cli_abort(
+      "Could not convert `country` '{country}' to an ISO 3 character \\
+      country code; please check `country`."
+    )
+  }
+
+  # abs path to data; currently only supporting local data
+  data_location <- file.path(data_location, country_iso3c)
+  if (dir.exists(data_location)) {
+    files <- list.files(data_location, full.names = TRUE)
+    data_source_id <- data_source_names[data_source]
+    search_pattern <- sprintf(
+      "%s_admin_%s.parquet",
+      data_source_id,
+      admin_level
+    )
+
+    # search for files
+    target_file <- files[grepl(search_pattern, files)]
+
+    # check that file is found
+    if (identical(target_file, character(0))) {
+      cli::cli_abort(
+        "File '{target_file}' not found; please check that data are present in
+        '{data_location}'"
+      )
+    }
+
+    # read files and return data.frame
+    data <- arrow::read_parquet(target_file)
+    data <- as.data.frame(data)
+
+    # filter on date, assume timezone is UTC
+    data$Date <- lubridate::as_date(data$Date)
+
+    data[
+      lubridate::`%within%`(
```

Comment:
Please don't use infix operators as infix - really lubridate should provide a prefix form of this, but I think it would be less alarming to use `@importFrom` above this function probably

That said, it might be easier just to use

```
data$Date >= min(date_range) & data$Date <= max(date_range)
```

which would make more explicit the inclusiveness (or not) of date ranges